static/js/controllers/configuration-upgrade.js

```diff
@@ -0,0 +1,110 @@
+var PouchDB = require('pouchdb-browser');
+
+// REVIEWER: are we OK hard-coding these two concepts? If not, where should
+// we put this? We don't have access to env vars in medic-webapp.
+var IS_PROD_URL = /^https:\/\/[^.]+.app.medicmobile.org\//,
+    BUILDS_DB = 'https://staging.dev.medicmobile.org/_couch/builds';
+
+angular.module('inboxControllers').controller('ConfigurationUpgradeCtrl',
+  function(
+    $log,
+    $q,
+    $scope,
+    DB
+  ) {
+
+    'use strict';
+    'ngInject';
+
+    $scope.loading = true;
+    $scope.versions = {};
+
+    DB().get('_design/medic')
+      .then(function(ddoc) {
+        $scope.deployInfo = ddoc.deploy_info;
+
+        $scope.allowBetas = $scope.allowBranches =
+          !window.location.href.match(IS_PROD_URL);
+
+        var buildsDb = new PouchDB(BUILDS_DB);
+
+        var viewLookups = [];
+
+        viewLookups.push(
+          buildsDb
+            .query('builds/branches')
+            .then(function(res) {
+              $scope.versions.branches = res.rows.sort(function(a, b) {
+                return Date.parse(b.value.build_time) -
+                       Date.parse(a.value.build_time);
```

Comment:
Sort on the server by keying the view by build_time, then include a limit in the query, otherwise this will eventually blow out.